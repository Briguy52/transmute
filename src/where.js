import curry from "./curry";
import filter from "./filter";
import match from "./match";

function where(pattern, subject) {
  return filter.operation(match(pattern), subject);
}

/**
 * Takes items in `subject` that match `pattern`.
 *
 * @param  {Function} pattern
 * @param  {Iterable} subject
 * @return {Iterable}
 */
export default curry(where);
