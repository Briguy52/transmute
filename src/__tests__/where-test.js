import { List, Map } from "immutable";
import where from "../where";

describe("transmute/where", () => {
  const mockColby = Map({
    name: "Colby",
    age: 24
  });
  const mockJill = Map({
    name: "Jill",
    age: 23
  });
  const mockShaun = Map({
    name: "Shaun",
    age: 23
  });
  const mockHayStack = List([mockColby, mockJill, mockShaun]);

  it("matches one item", () => {
    expect(where({ name: "Colby" }, mockHayStack)).toMatchSnapshot();
  });

  it("matches multiple items", () => {
    expect(where({ age: 23 }, mockHayStack)).toMatchSnapshot();
  });

  it("matches maps", () => {
    const subject = Map({
      colby: mockColby,
      jill: mockJill,
      shaun: mockShaun
    });
    expect(where({ age: 23 }, subject)).toMatchSnapshot();
  });
});
