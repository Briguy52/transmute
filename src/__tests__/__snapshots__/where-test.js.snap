// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transmute/where matches maps 1`] = `
Immutable.Map {
  "jill": Immutable.Map {
    "name": "Jill",
    "age": 23,
  },
  "shaun": Immutable.Map {
    "name": "Shaun",
    "age": 23,
  },
}
`;

exports[`transmute/where matches multiple items 1`] = `
Immutable.List [
  Immutable.Map {
    "name": "Jill",
    "age": 23,
  },
  Immutable.Map {
    "name": "Shaun",
    "age": 23,
  },
]
`;

exports[`transmute/where matches one item 1`] = `
Immutable.List [
  Immutable.Map {
    "name": "Colby",
    "age": 24,
  },
]
`;
